pkg/kubectl/cmd/run.go

```diff
@@ -60,6 +60,7 @@ func NewCmdRunContainer(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().Bool("dry-run", false, "If true, only print the object that would be sent, without sending it.")
 	cmd.Flags().String("overrides", "", "An inline JSON override for the generated object. If this is non-empty, it is used to override the generated object. Requires that the object supply a valid apiVersion field.")
 	cmd.Flags().Int("port", -1, "The port that this container exposes.")
+	cmd.Flags().Int("hostport", -1, "The host port mapping for the container port. To demonstrate a single-machine container.")
```

Comment:
Just prefixing the flag makes it awkward to use, especially in a demo. I'd
prefer to keep to limit qualifications in the comment line as long as we
don't deprecate it in the API.


On Wed, Apr 29, 2015 at 5:54 PM, Vish Kannan <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/run.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7536#discussion_r29396398>
> :
>
> > @@ -60,6 +60,7 @@ func NewCmdRunContainer(f *cmdutil.Factory, out io.Writer) *cobra.Command {
> >  	cmd.Flags().Bool("dry-run", false, "If true, only print the object that would be sent, without sending it.")
> >  	cmd.Flags().String("overrides", "", "An inline JSON override for the generated object. If this is non-empty, it is used to override the generated object. Requires that the object supply a valid apiVersion field.")
> >  	cmd.Flags().Int("port", -1, "The port that this container exposes.")
> > +	cmd.Flags().Int("hostport", -1, "The host port mapping for the container port. To demonstrate a single-machine container.")
>
> Should we explicitly name this flag as experimental or deprecated since
> we are adding only for demo purposes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7536/files#r29396398>
> .
>